src/js/rbac/fetchPermissions.js

```diff
@@ -23,3 +22,28 @@ export const fetchPermissions = (userToken, app = '') => {
         }})
     .catch(error => log(error));
 };
+
+export const createFetchPermissionsWatcher = (getUser, libjwt) => {
+    let currentCall = undefined;
+    const cache = new CacheAdapter(
```

Comment:
I wouldn't create new cache, I'd rather share chrome cache from all over places, meaning passing it to `createFetchPermissionsWatcher`
```JS
const createFetchPermissionsWatcher = (cache) => {
  let currentCall = undefined;
    return async (userToken, app = '') => {
        if (insights.chrome.getBundle() === 'openshift') {
            return Promise.resolve([]);
        }
        const permissions = await cache.getItem('permissions');
        if (permissions) {
            return permissions;
        }
        if (typeof currentCall === 'undefined') {
            currentCall = getUser().then(() => fetchPermissions(userToken, app).then((data) => {
                currentCall = undefined;
                cache.setItem('permissions', data);
                return data;
            }));
        }
        return currentCall;
    };
}
```

Reply:
@kate that would be perfect however that will take a little bit more effort than just lifting to a new file which will delay this fix. I would like to do that in a different PR.